### PR TITLE
Create indices with disabled analysis

### LINKED ISSUE #89: Make string fields not_analyzed in adapter-created indices
When the adapter creates an index, its string fields should be not_analyzed so you can do filters and aggregations on them.

### LINKED ISSUE #90: Don't write to indices / document types that don't exist
This is a sister issue to #89, both in service of the same goal: steer the user away from writing analyzed string fields to ES, because that won't result in good behavior when reading back through juttle (specifically, `name=value` filter and `reduce by` will not do what you'd want them to do). 

If the user set up the mapping in ES explicitly when creating their index and document type, we will trust them to know what they're doing. However, when using `write elastic`, they may assume that we are choosing the "right" setup. We do that for the default `juttle` index in #89, but if the user names a different index / doc type, we only have two choices:

1. if the index / doc type combination already exists, it must have been set up by the user explicitly as they desired, and we will proceed to write to it;
2. if the index / doc type combo does not exist, we should refuse to write, rather than going ahead with the write and getting the default all-strings-as-analyzed ES mapping.

That's what this issue is for: let's fail to write if the named index / type does not already exist. (For interval-based indices, it means that not a single instance of the index exists.)

### PR DESCRIPTION
**Pull request by davidvgalbraith**
fixes #90
fixes #89

### CODE REVIEW

**Comment by davidvgalbraith** [comment]
@VladVega @demmer

File: test/elastic-adapter.spec.js [hunk 1]
```diff
@@ -138,7 +160,7 @@ describe('elastic source', function() {
                 });
             });
 
-            it('writes a point with a giant field', function() {
+            it('fails to write a point with a giant field', function() {
```
Comment (dmehra) [inline test/elastic-adapter.spec.js]: This seems like an unrelated change

**Comment by dmehra** [comment]
I'd like to see tests that validate the behavior we want, which prompted this whole thing:
Write points with string value containing spaces or other non-word characters, such as `msg = 'hello world!'`. Then 1) read with matching filter, and 2) read | reduce by msg. And validate that FTS on full 'hello world!' and individual 'hello' and 'world' still works (this part may be already covered by existing test).

**Comment by dmehra** [comment]
Note, you need to say something like "fixes #91" in the description, otherwise the issue won't close when you merge. Also makes it easier for the reviewers who can then click on the issue link for context (can't click from the title).

File: test/elastic-adapter.spec.js (hunk 1 above)
Comment (davidvgalbraith) [inline test/elastic-adapter.spec.js]: It's related! It turns out we were able to write the big string because the analyzer broke it into teeny chunks, but without analysis it tries to store the whole thing which is too much.

Comment (dmehra) [inline test/elastic-adapter.spec.js]: Oic. Then how about a positive test that we can store max_allowed string? Maybe in the aptly named elastic-adapter-limits.spec.js

**Comment by davidvgalbraith** [comment]
I added the suggested tests.

**Comment by dmehra** [comment]
+1 on the positive tests. The test for writing a big field that's not too big still needs to be added, but orthogonal to this PR, let's add to the list in #88.

**Comment by VladVega** [comment]
So what are the implications of no-analyzing?

File: test/elastic-adapter-limits.spec.js [hunk 2]
```diff
@@ -22,7 +20,7 @@ describe('elastic source limits', function() {
             before(function() {
                 return retry(function() {
                     return test_utils.write(points, {id: type})
-                    .then(function() {
+                    .then(function(result) {
```
Comment (VladVega) [inline test/elastic-adapter-limits.spec.js]: did `lint` catch this?

**Comment by dmehra** [comment]
Let me see if i can channel my inner dave: No implications whatsoever, in our case!
Generally using not_analyzed fields means you cannot do terms queries (full text search, in juttle terminology) on those individual fields; however juttle doesn't currently support FTS on a specific field, so this is a moot point.
We can still do FTS across all fields as that's implemented using a magic _all field in ES which combines the values of all string fields (at the cost of storage).

File: lib/elastic.js [hunk 3]
```diff
@@ -11,6 +10,8 @@ var utils = require('./utils');
 var logger = require('juttle/lib/logger').getLogger('elastic');
 
 var instances;
+var ALREADY_CREATED_INDICES = {};
```
Comment (VladVega) [inline lib/elastic.js]: mmm. So this is not really a constant. Generally in the adapter only constants are defined in all caps?

File: lib/elastic.js [hunk 4]
```diff
@@ -116,10 +112,55 @@ function _get_mapping(client, index, type) {
...
+function _create_index(client, index, type) {
+    var options = {
+        index: index,
+        body: {
+            mappings: {}
+        }
+    };
+    options.body.mappings[type] = DYNAMIC_MAPPING_SETTINGS;
+    if (client instanceof AmazonElasticsearchClient) {
+        return client.createIndexAsync(options);
```
Comment (VladVega) [inline lib/elastic.js]: Just curious: instead of `if (client instanceof AmazonElasticsearchClient)` in a bunch of places, does it make sense for a client super class to figure this out and choose the correct function to call?

File: lib/elastic.js (hunk 3 above)
Comment (davidvgalbraith) [inline lib/elastic.js]: I capitalize all globals, like https://github.com/juttle/juttle-elastic-adapter/blob/master/lib/utils.js#L17.

File: lib/elastic.js (hunk 4 above)
Comment (davidvgalbraith) [inline lib/elastic.js]: It makes sense to have the `AmazonElasticsearchClient` have exactly the same API as the local Elasticsearch client, so nobody has to figure it out. That's my plan here.

File: lib/elastic.js (hunk 3 above)
Comment (davidvgalbraith) [inline lib/elastic.js]: Ludicrous! Let's downcase this.

File: lib/elastic.js (hunk 4 above)
Comment (VladVega) [inline lib/elastic.js]: Super.

**Comment by VladVega** [comment]
+1 once the client dichotomy is resolved (even if that means opening a ticket to fix it) while merging this.

**Comment by davidvgalbraith** [comment]
Opened https://github.com/juttle/juttle-elastic-adapter/issues/96.

File: test/elastic-adapter-limits.spec.js (hunk 2 above)
Comment (davidvgalbraith) [inline test/elastic-adapter-limits.spec.js]: Nope!
